qiskit/chemistry/drivers/fcidumpd/parser.py

```diff
@@ -0,0 +1,227 @@
+        hijkl_ab_elements = set(itertools.product(
+            range(norb), range(norb), beta_range, beta_range
+        ))
+        hijkl_ba_elements = set(itertools.product(
+            beta_range, beta_range, range(norb), range(norb)
+        ))
+        hijkl_bb_elements = set(itertools.product(beta_range, repeat=4))
+    orbital_data = fcidump_str[namelist_end.end(0):].split('\n')
+    for orbital in orbital_data:
+        if not orbital:
+            continue
+        x = float(orbital.split()[0])
+        # Note: differing naming than ijkl due to E741 and this iajb is inline with this:
+        # https://hande.readthedocs.io/en/latest/manual/integrals.html#fcidump-format
+        i, a, j, b = [int(i) for i in orbital.split()[1:]]  # pylint: disable=invalid-name
+        if i == a == j == b == 0:
+            output['ecore'] = x
+        elif a == j == b == 0:
+            # TODO: x is the energy of the i-th MO
+            continue
+        elif j == b == 0:
+            try:
+                hij_elements.remove((i-1, a-1))
+                hij[i-1][a-1] = x
+            except KeyError:
+                if _uhf:
+                    hij_b_elements.remove((i-1, a-1))
+                    hij_b[i-1-norb][a-1-norb] = x
+                else:
+                    raise QiskitChemistryError()
+        else:
+            try:
+                hijkl_elements.remove((i-1, a-1, j-1, b-1))
+                hijkl[i-1][a-1][j-1][b-1] = x
+            except KeyError:
+                if _uhf:
+                    try:
+                        hijkl_ab_elements.remove((i-1, a-1, j-1, b-1))
+                        hijkl_ab[i-1][a-1][j-1-norb][b-1-norb] = x
+                    except KeyError:
+                        try:
+                            hijkl_ba_elements.remove((i-1, a-1, j-1, b-1))
+                            hijkl_ba[i-1-norb][a-1-norb][j-1][b-1] = x
+                        except KeyError:
+                            hijkl_bb_elements.remove((i-1, a-1, j-1, b-1))
+                            hijkl_bb[i-1-norb][a-1-norb][j-1-norb][b-1-norb] = x
+                else:
+                    raise QiskitChemistryError()
+
+    # iterate over still empty elements in 1-electron matrix and populate with symmetric ones
+    # if any elements are not populated these will be zero
+    _permute_1e_ints(hij, hij_elements, norb)
+
+    if _uhf:
+        # do the same for beta spin
+        _permute_1e_ints(hij_b, hij_b_elements, norb, beta=True)
+
+    # do the same of the 2-electron 4D matrix
+    _permute_2e_ints(hijkl, hijkl_elements, norb)
+
+    if _uhf:
+        # do the same for beta spin
+        _permute_2e_ints(hijkl_bb, hijkl_bb_elements, norb, beta=2)
+        _permute_2e_ints(hijkl_ab, hijkl_ab_elements, norb, beta=1)
+        _permute_2e_ints(hijkl_ba, hijkl_ba_elements, norb, beta=1)
+
+        # assert that EITHER hijkl_ab OR hijkl_ba were given
+        assert np.allclose(hijkl_ab, 0.0) != np.allclose(hijkl_ba, 0.0)
```

Comment:
Preferably this would be a raise with a message so the user knows whats going wrong, is that possible?